Stats Revamp v2: Refresh data on insight details

ravishanker opened the pull request:
When going from insights cards to it's detail screen, sometimes the cards are not displaying immediately


To test:

- Launch Jetpack app
- Navigate to Stats 
- On Insights tab, tap on View More on Views & Visitors or Total Likes or Total Comments or Total Followers cards
- Ensure it opens respective details screen
- On the respective details screen, all the related cards are displayed if data exists.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested with different use cases

3. What automated tests I added (or what prevented me from doing so)
existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Refresh data on insight details

When going from insights cards to it's detail screen, some times the cards are not displaying immediately

AliSoftware commented:
I'm doing code freeze of 20.2 today, so it is quite too late for this brand new PR to be part of `20.2`. As such, I'm updating its milestone to `20.3`.

irfano committed:
Revert "Refresh data on insight details"

This reverts commit f3e352aa1278f6e155c86271d547b302894c422e.

irfano committed:
Update stats tab offscreenPageLimit to fix datepicker problem

Views & Visitors and Total Likes detail screen stuck at loading state in some cases. offscreenPageLimit is increased as a workaround.

irfano commented:
@ravishanker, the problem was still there with your commit. I reverted it. It didn't help, and it may cause extra redundant calls. If you think it solved another problem, we can add it again.

### Problem
The screen was stuck at the loading state on the Views & Visitors, Total Likes detail screens. The reason could be this,`selectedDate` is null here on these screens:
https://github.com/wordpress-mobile/WordPress-Android/blob/898f0354acb7bbb2a25438e1f95c40c4492d4c25/WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/granular/GranularStatefulUseCase.kt#L31-L34
When selecting other tabs on the Insights screen, you cause WEEKS tabs caching. Then Views & Visitors works since Views & Visitors detail uses weeks granularity, and it can take advantage of WEEKS tabs caching on data selector.

### Solution
I increased stats tab `offscreenPageLimit` parameter. d954ec0c1f529b210b6e9201653e0f794445bc36 Now, when you open stats screen, WEEKS tab is being loaded and detail screens work weel.
This is a quick solution: we can call it a workaround. But it doesn't harm anything. In addition, the WEEKS tab will be in cache now, and opening faster at the first launch. 😄

ravishanker commented:
No longer required
